### ISSUE TITLE
Add a new type for slow motion videos

### ISSUE THREAD

**Issue by xmha97** [deckerst/aves#585]
Hi, how I can filter for slow motion videos?

![Screenshot_2023-04-14-08-32-38-141_deckers.thibault.aves.jpg](https://user-images.githubusercontent.com/23485114/231946129-5035a9cb-abf8-4345-8ff5-40a9a1298b5c.jpg)

**Comment by deckerst** [comment]
What is a slow motion video? Does it have some specific metadata? Please share a sample.

**Comment by xmha97** [comment]
> What is a slow motion video? Does it have some specific metadata? Please share a sample.

I think slow motion videos are 120 fps without any specific metadata:

![Screenshot_2023-04-14-10-11-41-976_deckers.thibault.aves.jpg](https://user-images.githubusercontent.com/23485114/231965736-487aaf72-d57d-4d58-96ac-ff8a38925aea.jpg)

A sample video:

https://www.dropbox.com/s/63v39ducoszxwxo/VID_20230414_031236_HSR_120.mp4?dl=1

**Comment by xmha97** [comment]
Thanks for your review, I'll try to check other devices to see if we can find a solution.

**Comment by xmha97** [comment]
Tests on Samsung S21 FE

Samsung Pro Video FHD 120FPS
https://www.dropbox.com/s/5si70c6n6p0j1i6/Samsung%20Pro%20Video%20FHD%20120FPS.mp4?dl=1

Samsung Super Slow Motion 960FPS
https://www.dropbox.com/s/ngxseqytiqmpgos/Samsung%20Super%20Slow%20Motion%20960FPS.mp4?dl=1

Samsung Slow Motion 240FPS
https://www.dropbox.com/s/5knz7cjzrtbeame/Samsung%20Slow%20Motion%20240FPS.mp4?dl=1

**Comment by deckerst** [comment]
| Video                                                  | Capture FPS | Stream FPS | Slow mo | Aves | GPhotos | Xiaomi Gal. | Samsung Gal. |
| -----------------------------------------------  |------------------ | ---------------- | ------------- | ------- | ------------- | ---------------- | -------------------- |
| Pixel Slow Motion 120FPS              | 120.0             | 30.075         | ✅            | ✅     | ✅            | ❓                | ❓                     |
| Pixel Slow Motion 240FPS              | 240.0             | 30.078         | ✅            | ✅     | ✅            | ❓                | ❓                     |
| Samsung Pro Video FHD 120FPS  | 120.0             | 120.019       | ❌            | ❌     | ❌            | ✅                | ❌                     |
| Samsung Super Slow Mo 960FPS | undefined    | 30.0              | ✅            | ❌     | ❌            | ❌                | ✅                     |
| Samsung Slow Motion 240FPS      | 240.0             | 29.884         | ✅            | ✅     | ✅            | ❌                | ✅                     |
| Xiaomi Slow Motion 120FPS          | 120.0             | 119.346       | ✅            | ✅     | ❌            | ✅                | ❌                     |

edit: as of 2026/06/09, Google Photos does not detect the same slow motion videos as it did in 2023.

**Comment by deckerst** [comment]
Apparently there's no standard, different manufacturers have different interpretations, and I don't see any characteristic metadata (besides `com.xiaomi.slow_moment`). The capture rate could be used, though:
- if I identify a slow-mo video as one with 120 FPS, it will give the same ID as Xiaomi Gallery,
- if I identify a slow-mo video as one with at least 120 FPS, it will give the same ID as Google Photos.

Honestly I don't know whether it's worth it, because that kind of special videos are features quickly developed and quickly forgotten by manufacturers. Also, even if I add the identification, I'm unlikely to allow playback tuning as requested in #583 because the current video engine used by Aves is not so performant/convenient.

**Comment by xmha97** [comment]
Hi Thibault
Thank you for your explanation and time on this issue.

I downloaded some slow-motion videos from several Telegram groups. (Modified versions of Google Camera are discussed in these groups)

These videos are displayed in Google Photos as slow-motion videos:
https://www.dropbox.com/s/bknq94b3a6w998f/Fan.mp4?dl=1
https://www.dropbox.com/s/s3uobuunrw8j1sx/Flower.mp4?dl=1
https://www.dropbox.com/s/yuy4gi7abohmg6h/Train.mp4?dl=1

This video is displayed in the Xiaomi gallery as a slow-motion video:
https://www.dropbox.com/s/enxtv2pag7yybuo/PowerStrip.mp4?dl=1

I hope the following conclusions are correct:

Xiaomi only checks the `frame rate` tag, on my brother's Samsung phone, I took a 120 fps video (which was not slow motion) but Xiaomi displayed it as a slow motion video!
I downloaded a 120 fps video (which I think does not have any special metadata) and saw that Xiaomi also recognized it as a slow-motion video!
https://archive.org/download/4-k-120fps/%E9%9B%AA%E5%9C%B0%204K%40120fps.mp4
Please don't follow Xiaomi's method to detect slow-motion videos, although I use Xiaomi devices.

I think the standard of Google and Samsung is the same because Samsung's slow-motion videos are displayed in Google Photos as a slow-motion videos.

Not so in the case of Xiaomi, Xiaomi's slow-motion videos are not recognized as slow-motion videos in Google Photos.

I looked at the tags of the videos and noticed an issue.
The videos taken by the Xiaomi camera, only have the `frame rate` tag.

![image](https://user-images.githubusercontent.com/23485114/232469292-0e305c98-c803-4417-a465-960b1b123aff.png)

But in the videos that are saved by Google Camera and Samsung Camera, in addition to the `frame rate` tag, there is also the `real frame rate` tag.

![image](https://user-images.githubusercontent.com/23485114/232469064-5cb0845e-d5a6-47c1-96bd-a6eb0e53fac8.png)

I think the detection is done through these two tags, although on the computer I am not able to make a video whose `frame rate` tag is different from the `real frame rate` tag and I can't understand the detection method.

Hi Thibault
Thank you for your explanation and time on this issue.

I downloaded some slow-motion videos from several Telegram groups. (Modified versions of Google Camera are discussed in these groups)

These videos are displayed in Google Photos as slow-motion videos:
https://www.dropbox.com/s/bknq94b3a6w998f/Fan.mp4?dl=1
https://www.dropbox.com/s/s3uobuunrw8j1sx/Flower.mp4?dl=1
https://www.dropbox.com/s/yuy4gi7abohmg6h/Train.mp4?dl=1

This video is displayed in the Xiaomi gallery as a slow-motion video:
https://www.dropbox.com/s/enxtv2pag7yybuo/PowerStrip.mp4?dl=1

I hope the following conclusions are correct:

Xiaomi only checks the `frame rate` tag, on my brother's Samsung phone, I took a 120 fps video (which was not slow motion) but Xiaomi displayed it as a slow motion video!
I downloaded a 120 fps video (which I think does not have any special metadata) and saw that Xiaomi also recognized it as a slow-motion video!
https://archive.org/download/4-k-120fps/%E9%9B%AA%E5%9C%B0%204K%40120fps.mp4
Please don't follow Xiaomi's method to detect slow-motion videos, although I use Xiaomi devices.

I think the standard of Google and Samsung is the same because Samsung's slow-motion videos are displayed in Google Photos as a slow-motion videos.

Not so in the case of Xiaomi, Xiaomi's slow-motion videos are not recognized as slow-motion videos in Google Photos.

I looked at the tags of the videos and noticed an issue.
The videos taken by the Xiaomi camera, only have the `frame rate` tag.

![image](https://user-images.githubusercontent.com/23485114/232469292-0e305c98-c803-4417-a465-960b1b123aff.png)

But in the videos that are saved by Google Camera and Samsung Camera, in addition to the `frame rate` tag, there is also the `real frame rate` tag.

![image](https://user-images.githubusercontent.com/23485114/232469064-5cb0845e-d5a6-47c1-96bd-a6eb0e53fac8.png)

I think the detection is done through these two tags, although on the computer I am not able to make a video whose `frame rate` tag is different from the `real frame rate` tag and I can't understand the detection method.

**Comment by deckerst** [comment]
Thanks for digging into this.

You're right, this `real frame rate` could be a good indicator to identify slow-motion videos. Unfortunately, I don't know yet how I can read it (or derive it). In the [table above](https://github.com/deckerst/aves/issues/585#issuecomment-1510350218), I wrote the `capture frame rate` and `stream frame rate`, which already show some differences. But it's not exact science, as you can see the capture rate is missing for the super slow-mo 960FPS Samsung video.

**Comment by xmha97** [comment]
> Unfortunately, I don't know yet how I can read it (or derive it).

It is possible through Exiftool, although I know it is not useful for you:
![image](https://user-images.githubusercontent.com/23485114/232496422-ba1ad2a4-963d-4fe8-8eeb-fd4783f449d5.png)

> But it's not an exact science, as you can see the capture rate is missing for the super slow-mo 960FPS Samsung video.

Samsung's Super Slow Motion is a normal video and nowhere is it identified as a specific video.
This video is an edited video, the beginning and the end are normal videos and the middle is slow motion.
Please do not consider this.

**Comment by xmha97** [comment]
Do not pay attention to this part, maybe it will be useful in the future:

# Time Lapse

Xiaomi Camera
```
com.xiaomi.fast_moment : 1
```

Google Camera
```
SpecialTypeID : com.google.android.apps.camera.gallery.specialtype.SpecialType-TIMELAPSE
```

Samsung Camera
```
N/A
```

![image](https://user-images.githubusercontent.com/23485114/233788832-72a1ad7c-c366-4f1f-9bbb-9409d0223e33.png)

**Comment by deckerst** [comment]
Did you send me a sample for the Google Camera timelapse video? You talk about it but I can't find it among my samples.

**Comment by Jakarrrg** [comment]
Thank you for reopening it. I have a Google Pixel 6 pro. I am willing to help by providing sample videos, and whatever I can offer, although I am not a programmer myself. Should I just shoot some slow motion videos and upload them here from the "add files" section? Is there something specific that would help?

**Comment by deckerst** [comment]
@Jakarrrg just a sample from you would be nice, thank you. I already have some others from Xiaomi and Samsung (thanks to the people in this thread), and having one from a Pixel would confirm the landscape.

**Comment by Jakarrrg** [comment]
https://wormhole.app/l89JE1#0DZzfMByIIfyI4FqmKPSuQ  (24h - 5 downloads link) 

I uploaded two video samples here. The one is 1/4 speed and the other 1/8. 

I use Pixel 6 pro with GrapheneOS without gms, but I have installed a custom pixel camera app (έκδοση 9.4.103.650360311.25
com.android.MGC_9_4_103
versionCode 67635286
targetSdk 34
minSdk 30), and have enabled the "special access to hardware accelerators to Google apps", so I believe the samples should be valid.
